app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -126,7 +170,89 @@ public CharSequence getPageTitle(int position) {
 
         @Override
         public int getCount() {
-            return this.tabTitles.length;
+            if(PreferenceManager.getDefaultSharedPreferences(getActivity())
+                    .getString(getString(R.string.main_page_content_key), getString(R.string.blank_page_key))
+                    .equals(getString(R.string.subscription_page_key))) {
+                return 1;
+            } else {
+                return 2;
+            }
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Main page content
+    //////////////////////////////////////////////////////////////////////////*/
+
+    private Fragment getMainPageFramgent() {
```

Comment:
Must we hard-code the strings in this method? If so, we should extract them.

Reply:
No, we discused about that. We came to conclution to not do that.

Reply:
I meant extracting the hardcoded strings and urls to a static variable or maybe put them in the Constants file.

Reply:
Ow I see.

Reply:
I created some static variables in the header instead

Reply:
fixed